### PR TITLE
HW2 code

### PR DESCRIPTION
**Pull request by ksing**
HW1 files were moved to a new folder - hw1. 
HW2 files are in a new folder hw2

### CODE REVIEW

File: course7/homeworks/kushagra/hw2/ls_hw2.cpp [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    }
+}
+
+
+int main(int argc, char** argv)
+{
+	std::string path_name = ".";
+	bool details = false, recursive = false;
+	
+    po::options_description desc("Display contents of a directory.");
```
Comment (maciekgajewski) [inline course7/homeworks/kushagra/hw2/ls_hw2.cpp:113]: Inconsistent indentation

File: course7/homeworks/kushagra/hw2/ls_hw2.cpp [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+			"Recursively display directory contents");
+	po::positional_options_description posOption;
+    posOption.add("path_name", -1);
+    po::variables_map vm;
+	po::store(po::command_line_parser(argc, argv).options(desc).positional(posOption).run(), vm);
+	po::notify(vm);
+
+	if(vm.count("path_name")) {
+		path_name = vm["path_name"].as<std::string>();
+		if (path_name.back() == '/') {
```
Comment (maciekgajewski) [inline course7/homeworks/kushagra/hw2/ls_hw2.cpp:129]: is this really required?

File: course7/homeworks/kushagra/hw2/ls_hw2.cpp [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+	}
+	if(vm.count("long")) {
+		details = vm["long"].as<bool>();
+	}
+	if(vm.count("tee")) {
+		recursive = vm["tee"].as<bool>();
+	}
+	if(vm.count("help")) {
+        std::cout << desc << std::endl; 
+       return 0;
```
Comment (maciekgajewski) [inline course7/homeworks/kushagra/hw2/ls_hw2.cpp:141]: wrong indentation again, here and in other places.

**Review by maciekgajewski — APPROVED** [review]
Good, only ugly. The indentation is messed up, which makes the code hard to read.

File: course7/homeworks/kushagra/hw2/ls_hw2.cpp (hunk 1 above)
Comment (ksing) [reply course7/homeworks/kushagra/hw2/ls_hw2.cpp:113]: Indeed. I need to fix this indentation, I think it's coming from me doing half of the homework at home, and other at the office.

File: course7/homeworks/kushagra/hw2/ls_hw2.cpp (hunk 2 above)
Comment (ksing) [reply course7/homeworks/kushagra/hw2/ls_hw2.cpp:129]: Maybe I should try to handle this better - I was getting an extra '/' while printing start directory names, if the input was a path to a directory, and contained a '/' at the end. I chose to remove it like this, but I suppose this maybe an unnecessary check.

**Pull request merged**